GoMeet/src/main/webapp/js/manual-time-proposal.js

```diff
@@ -0,0 +1,161 @@
+  inputField.type = 'datetime-local';
+  inputField.name = MEETING_TIME_NAME_PREFIX + timeNameSuffix.toString();
+  // restrict potential meeting times to future dates only
+  let minDate = getDatetimeNow();
+  inputField.min = minDate;
+  // whenever user changes the input value, check what the time they inputted to be valid
+  inputField.onchange = function() {rectifyInputtedTime(this);};
+
+  let label = document.createElement('label');
+  label.htmlFor = inputField.name;
+  label.innerText = 'New Time';
+
+  let deleteButton = document.createElement('button');
+  deleteButton.innerText = '-';
+  deleteButton.className = 'delete-time-button';
+  deleteButton.onclick = function() {deleteTimeInput(document, inputField.name);}; 
+  
+  let enclosingDiv = document.createElement('div');
+  enclosingDiv.id = inputField.name; // ID of div = name of input field, to faciliate deleting
+
+  enclosingDiv.appendChild(label);
+  enclosingDiv.appendChild(inputField);
+  enclosingDiv.appendChild(deleteButton);
+  inputDiv.appendChild(enclosingDiv);
+}
+
+/**
+ * Deletes the input div which encloses the datetime-input <input> element, the 
+ * <label> for the <input> and the '-' delete button.
+ * @param {Document} document the document that this function operates on.
+ * @param {string} inputDivId the id of the div to delete. If invalid or not provided,
+ * this function will do nothing.
+ */
+function deleteTimeInput(document, inputDivId) {
+  if (inputDivId === null) { // No input ID given
+    return;
+  }
+  let inputDiv = document.getElementById(inputDivId);
+  if (inputDiv == null) { // Input ID is invalid (e.g. does not exist)
+    return;
+  }
+  let inputField = inputDiv.children.item(TIME_INPUT_FIELD_INDEX);
+  enteredTimes.delete(inputField.value);
+  inputDiv.remove();
+  timeInputs--; 
+  // Hide delete buttons if they only have one time input left
+  if (timeInputs === 1) {
+    toggleDeleteButtons(document);
+  }
+}
+
+/**
+ * Hides or shows the buttons with className 'delete-time-button'.
+ * Shows buttons if they're hidden, and vice-versa.
+ * Hides buttons by setting buttonElement.style.display to 'none', 
+ * and shows the buttons by setting buttonElement.style.display to
+ * 'inline'.
+ * If no buttons on the page were found with className 'delete-time-button',
+ * this function will do nothing.
+ * @param {Document} document the document that this function operates on.
+ */
+function toggleDeleteButtons(document) {
+  let deleteButtons = document.getElementsByClassName('delete-time-button');
+  for (let i = 0; i < deleteButtons.length; i++) {
+    if (deleteButtons[i].style.display === 'none') {
+      deleteButtons[i].style.display = 'inline';
+    } else {
+      deleteButtons[i].style.display = 'none';
+    }
+  }
+}
+
+/**
+ * Returns the ISO datestring of the local current date and time (hours and minutes),
+ * with the time zone data trimmed.
+ * Seconds and milliseconds are set to 0.
```

Comment:
Why are you trimming the time zone and minutes/seconds?

Reply:
The time zone is _always_ UTC time. So if I create a Date using `Date(datetimeStr with a time in AEDT)`, the ISO string of that date returned by Date.toISOString() will add a 'Z' to the end even though I meant to have the time in AEDT. Also, the 'min' property of the datetime input can only take in strings without the timezone. 

In terms of the seconds and milliseconds, if they are not set to 0, the datetime input field would include the seconds and milliseconds in the display, which looked a little bit messy. I also thought it would be confusing for user since I don't intend for them to be able set a meeting time to that level of granularity.

Reply:
I've updated the comments of the function to document this 😊